skyportal/models.py

```diff
@@ -3120,6 +3126,327 @@ class SourceNotification(Base):
+        if elem is None:
+            return None
+        else:
+            try:
+                return elem.attrib.get('value', '').replace('http://', 'https://')
+            except Exception:
+                return None
+
+    @property
+    def gracesa(self):
+        """Event page URL."""
+        try:
+            notice = self.gcn_notices[0]
+        except IndexError:
+            return None
+        root = lxml.etree.fromstring(notice.content)
+        elem = root.find(".//Param[@name='EventPage']")
+        if elem is None:
+            return None
+        else:
+            try:
+                return elem.attrib.get('value', '')
+            except Exception:
+                return None
+
+    @property
+    def ned_gwf(self):
+        """NED URL."""
+        return "https://ned.ipac.caltech.edu/gwf/events"
+
+    @property
+    def HasNS(self):
+        """Checking if GW event contains NS."""
+        notice = self.gcn_notices[0]
+        root = lxml.etree.fromstring(notice.content)
+        elem = root.find(".//Param[@name='HasNS']")
+        if elem is None:
+            return None
+        else:
+            try:
+                return 'HasNS: ' + elem.attrib.get('value', '')
+            except Exception:
+                return None
+
+    @property
+    def HasRemnant(self):
+        """Checking if GW event has remnant matter."""
+        notice = self.gcn_notices[0]
+        root = lxml.etree.fromstring(notice.content)
+        elem = root.find(".//Param[@name='HasRemnant']")
+        if elem is None:
+            return None
+        else:
+            try:
+                return 'HasRemnant: ' + elem.attrib.get('value', '')
+            except Exception:
+                return None
+
+    @property
+    def FAR(self):
+        """Returning event false alarm rate."""
+        notice = self.gcn_notices[0]
+        root = lxml.etree.fromstring(notice.content)
+        elem = root.find(".//Param[@name='FAR']")
+        if elem is None:
+            return None
+        else:
+            try:
+                return 'FAR: ' + elem.attrib.get('value', '')
+            except Exception:
+                return None
+
+
+class GcnNotice(Base):
+    """Records of ingested GCN notices"""
+
+    id = sa.Column(sa.Integer, primary_key=True, autoincrement=True)
+
+    ivorn = sa.Column(
+        sa.String, primary_key=True, comment='Unique identifier of VOEvent'
+    )
+
+    notice_type = sa.Column(
+        sa.Enum(gcn.NoticeType, native_enum=False),
+        nullable=False,
+        comment='GCN Notice type',
+    )
+
+    stream = sa.Column(
+        sa.String, nullable=False, comment='Event stream or mission (i.e., "Fermi")'
+    )
+
+    date = sa.Column(sa.DateTime, nullable=False, comment='UTC message timestamp')
+
+    dateobs = sa.Column(
+        sa.DateTime,
+        sa.ForeignKey(GcnEvent.dateobs),
+        nullable=False,
+        comment='UTC event timestamp',
+    )
+
+    content = deferred(
```

Comment:
Nice, good idea to use `deferred` for these big columns

Reply:
+1!